### ISSUE TITLE
🐛 [BUG] Incorrect value for `--treeViewItem-leadingVisual-iconColor-rest` in dark high contrast mode

### ISSUE THREAD

**Issue by ktravers** [primer/primitives#1083]
#### TL;DR

In dark high contrast mode, the current value of `--treeViewItem-leadingVisual-bgColor-rest` (`#b7bdc8`) is incorrect. We need to update the value to the correct value, `#f0f3f6`.

#### Full story

While working on https://github.com/primer/react/pull/5267, I ran into unexpected visual regression test failures. Specifically, the visual regression tests were failing in dark high contrast mode only because the expected value for the TreeView.Item LeadingVisual icon fill was different than the actual value.

Prior to https://github.com/primer/react/pull/5267, the icon fill was set by [custom theming within primer/react](https://github.com/primer/react/blob/21adedbc95e39a1b62ca2e0c31e87ac81e11911b/packages/react/src/legacy-theme/ts/color-schemes.ts#L3535) that referenced a deprecated variable and a hard-coded fallback value:

```json
"treeViewItem": {
  "directory": {
    "fill": "var(--treeViewItem-leadingVisual-bgColor-rest, var(--color-tree-view-item-directory-fill, #f0f3f6))"
  }
},
```

The variable `--treeViewItem-leadingVisual-bgColor-rest` was replaced by `--treeViewItem-leadingVisual-iconColor-rest` in https://github.com/primer/primitives/pull/686. When that replacement was made, we didn't update the theme in primer/react. As a result, when we calculate the `fill` value, it always uses the fallback, because the given variables no longer exist. This fallback is technically the "correct" color for dark high contrast mode.

Currently, `--treeViewItem-leadingVisual-bgColor-rest` is using a different value in dark high contrast mode, `#b7bdc8`. This value is incorrect. It should be using `#f0f3f6` instead. We [updated the primer/react snapshots](https://github.com/primer/react/pull/5267/commits/bdd9ffd435ce7637f5ebd28f184997a9b2181b74) as a temporary workaround, but now that https://github.com/primer/react/pull/5267 has shipped, we should update the variable to the correct value.

### Reproduction steps

Review vrt test report: [vrt-all-flags.zip](https://github.com/user-attachments/files/17783325/vrt-all-flags.zip)

### Expected behavior

TreeView.Item leading visual icon fill is `#f0f3f6` in dark high contrast mode.

### Screenshots

Correct directory icon fill | Current incorrect directory icon fill
--|--
![Screenshot of tree view with correct fill](https://github.com/user-attachments/assets/c3a13bbf-8473-4e83-bbc1-851e97dd7826) | ![Screenshot of tree view with incorrect fill](https://github.com/user-attachments/assets/fc86ffa2-061b-413a-a089-dd0925cebb9d)

### References

* https://github.com/primer/react/pull/5267
* https://github.com/primer/primitives/pull/686

/cc @langermank

**Comment by lukasoppermann** [comment]
Hey @ktravers,
can you explain **why** you feel that `#f0f3f6` is the correct value?

It is a custom value not available in our tokens. Whereas `#b7bdc8` is `neutral-10` so part of our scale.

The contrast is >10 so should be fine as well.

 ![Image](https://github.com/user-attachments/assets/a03b1a46-7f5b-4ffd-876f-1099a9ba4aa1)

**Comment by langermank** [comment]
@lukasoppermann I think its just meant to be `white` in dark high contrast. The exact hex value @ktravers provided is probably from the old neutral scale because we're basing this off of an older VRT snapshot image. So basically, we should just increase the contrast a bit on this token do it matches up better with the previous experience.

**Comment by langermank** [comment]
Just taking another look at this and adding Storybook tests for it, I think the current color value is totally fine. We haven't gotten any complains to my knowledge about the updated color and it aligns with the other neutral changes we've made to our high contrast themes. Closing this out 😄
